pulsar-testclient/src/main/java/org/apache/pulsar/testclient/PerformanceTransaction.java

```diff
@@ -0,0 +1,612 @@
+                        }
+                    }
+                }
+                for (String topic : arguments.consumerTopic) {
+                    log.info("Creating  consume partitioned topic {} with {} partitions", topic, arguments.partitions);
+                    try {
+                        client.topics().createPartitionedTopic(topic, arguments.partitions);
+                    } catch (PulsarAdminException.ConflictException alreadyExists) {
+                        if (log.isDebugEnabled()) {
+                            log.debug("Topic {} already exists: {}", topic, alreadyExists);
+                        }
+                        PartitionedTopicMetadata partitionedTopicMetadata =
+                                client.topics().getPartitionedTopicMetadata(topic);
+                        if (partitionedTopicMetadata.partitions != arguments.partitions) {
+                            log.error(
+                                    "Topic {} already exists but it has a wrong number of partitions: {}, expecting {}",
+                                    topic, partitionedTopicMetadata.partitions, arguments.partitions);
+                            PerfClientUtils.exit(-1);
+                        }
+                    }
+                }
+            }
+        }
+
+
+        PulsarClient client =
+                PulsarClient.builder().enableTransaction(arguments.isEnableTransaction).serviceUrl(arguments.serviceURL)
+                        .connectionsPerBroker(arguments.maxConnections)
+                        .ioThreads(arguments.ioThreads)
+                        .statsInterval(0, TimeUnit.SECONDS)
+                        .ioThreads(arguments.ioThreads).build();
+
+        ProducerBuilder<byte[]> producerBuilder = client.newProducer()//
+                .sendTimeout(0, TimeUnit.SECONDS);
+
+        final List<Future<Producer<byte[]>>> producerFutures = Lists.newArrayList();
+
+
+        ConsumerBuilder<byte[]> consumerBuilder = client.newConsumer(Schema.BYTES) //
+                .subscriptionType(arguments.subscriptionType)
+                .receiverQueueSize(arguments.receiverQueueSize)
+                .subscriptionInitialPosition(arguments.subscriptionInitialPosition)
+                .subscriptionInitialPosition(SubscriptionInitialPosition.Earliest);
+
+
+        Iterator<String> consumerTopicIterator = arguments.consumerTopic.iterator();
+        Iterator<String> producerTopicIterator = arguments.producerTopic.iterator();
+
+        final List<List<Consumer<byte[]>>> consumers = Lists.newArrayList();
+        for (int i = 0; i < arguments.numTestThreads; i++) {
+            final List<Consumer<byte[]>> subscriptions = Lists.newArrayListWithCapacity(arguments.numSubscriptions);
+            final List<Future<Consumer<byte[]>>> subscriptionFutures =
+                    Lists.newArrayListWithCapacity(arguments.numSubscriptions);
+            if (consumerTopicIterator.hasNext()) {
```

Comment:
In my understanding, each thread needs to create a corresponding number of consumers and producers. Maybe it should be a while loop.